README.rst

```diff
@@ -16,6 +9,46 @@ Purpose
 
 Redis Streams implementation for the Open edX event bus.
 
+Overview
+********
+This package implements an event bus for Open EdX using Redis streams.
+
+The event bus acts as a broker between services publishing events and other services that consume these events.
+
+This package contains both the publishing code, which processes events into messages to send to the stream, and the consumer code,
+which polls the stream using a `while True` loop in order to turn messages back into event data to be emitted.
+The actual Redis host is configurable.
+
+The repository works together with the openedx/openedx-events repository to make the fully functional event bus.
+
+Documentation
+*************
+
+To use this implementation of the Event Bus with openedx-events, you'll need to ensure that below the following Django settings are set::
+
+    # redis connection url
+    EVENT_BUS_REDIS_CONNECTION_URL: ...
+    EVENT_BUS_TOPIC_PREFIX: ...
+
+    # Required, on the producing side only:
+    EVENT_BUS_PRODUCER: edx_event_bus_redis.create_producer
+    # Required, on the consumer side only:
+    EVENT_BUS_CONSUMER: edx_event_bus_redis.RedisEventConsumer
+
+Optional settings that are worth considering::
+
+    # If the consumer encounters this many consecutive errors, exit with an error. This is intended to be used in a context where a management system (such as Kubernetes) will relaunch the consumer automatically.
+    EVENT_BUS_REDIS_CONSUMER_CONSECUTIVE_ERRORS_LIMIT (defaults to None)
+
+    # How long the consumer should wait, in seconds, for the Redis broker
```

Comment:
Can you explain what will happen when this timeout is hit?

Reply:
Updated the docs as well as default value for this setting from 1 second to 60 seconds as it makes no sense to run the loop every second even if no event is triggered. Let me know if you want to increase/decrease the default value.

Reply:
@bmtcril Yes, that is correct.